templates/tls-init-clusterrole.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) }}
+{{- if .Values.global.tls.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "consul.fullname" . }}-tls-init
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  annotations:
+    "helm.sh/hook": pre-install
```

Comment:
You didn't need to because acl-init is a post-install,post-upgrade hook, and so it's executed after all resources are created. This is not the case for pre-install hooks. They are executed before any resources are created by Helm, so we need to mark these too to make sure they are there.